Test.C

```diff
@@ -140,36 +144,37 @@ int main(){
         for (i = 0; i < NUM_EQUIPMENT; i++){
             // details about the equipment type itself is stored in equipmentStock
             // amount of this equipment type that the player has is stored in equipmentInventory
-            printf("%s: %d\n", equipmentStock[i]->name, player.equipmentInventory[i]);
+            printf("%s: %d\n", equipmentStock[i].name, player.equipmentInventory[i]);
         } //end for loop
 
-	      //pulls from info from index amount table referencing index struct table
-
+        //pulls from info from index amount table referencing index struct table
 
-        struct EncryptionInventory encryptionPointer = *player.encryptionInventory;
-	      while (encryptionPointer != null){
+        while (encryptionPtr != NULL) {
             // details about the encryption type itself is stored in encryptionStock
             // amount of this encryption type that the player has is stored in encryptionInventory
-            printf("%s\n", encryptionPointer->encryption->name); //current encrypt
-            encryptionPointer=encryptionPointer->next; //pointer
+            printf("%s\n", encryptionPtr->encryption.name); //current encrypt
+            encryptionPtr = encryptionPtr->next; //pointer
         } //end while loop
-		
-        //iterate and print through current encryption
-        
-    //receive credits
-		int sum = 0;
+
+        //receive credits
 		int newCredits = 0;
 		printf("Original credits: %d\n", player.currentCredits); //prints out the current credits the player has
 
-		
 		for (int i = 0; i < NUM_EQUIPMENT; i++){
-			sum += equipmentStock[i].messageAmount * player.equipmentInventory[i]; //sum message amount of equipment(s)
+            encryptionPtr = &player.encryptionInventory; // resets 'current' pointer
+            int currentLevel = 0;
+            double encryptionEffectiveness = 0.0;
+            while (encryptionPtr != NULL) { // traverses encryptionInventory looking for highest level allowed for equipment (assumes ordered list)
+                //if the encryption level is not allowed for the equipment, end the loop
+                if (encryptionPtr->encryption.level > equipmentStock[i].maxEncryptionLevel) {
+                    break;
+                }
+                encryptionEffectiveness = encryptionPtr->encryption.effectiveness;
+                encryptionPtr = encryptionPtr->next;
+            }
```

Comment:
This loop is entirely functional, but I think it makes more assumptions then necessary. You are assuming not only an ordered list, but that a newer encryption method will always be more effective than an older one. I suggest iterating through all encryption methods, checking if they are within the level range, and then if their effectiveness is greater than the current max.
```suggestion
            while (encryptionPtr != NULL) { // traverses encryptionInventory looking for highest level allowed for equipment 
                // ensure encryption method is available for this type of equipment
                if (encryptionPtr->encryption.level <= equipmentStock[i].maxEncryptionLevel){
                    // check if encryption method effectiveness is greater than the previous max
                    if (encryptionPtr->encryption.effectiveness > encryptionEffectiveness){
                        encryptionEffectiveness = encryptionPtr->encryption.effectiveness;
                    }
                }
                
                encryptionPtr = encryptionPtr->next;
            }
```